### PR TITLE
sql, TOC: split information schema from system database and improve

### PR DESCRIPTION
**Pull request by dcalvin**
In line with https://github.com/pingcap/docs/pull/994/

### CODE REVIEW

File: sql/information-schema.md [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+SHOW COLUMNS FROM table_name [FROM db_name] [LIKE 'wild']
+```
+
+### COLUMN\_PRIVILEGES Table
+
+空表。
+
+### ENGINES Table
+
+ENGINES 表提供了关于存储引擎的信息。从和 MySQL 兼容性上考虑，TiDB `show engines` 的结果展示成了 InnoDB。
```
Comment (lilin90) [inline sql/information-schema.md]: Please check all the following content and update it according to the English version.

File: sql/information-schema.md [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+| latin1             | latin1_swedish_ci    | cp1252 West European  |      1 |
+| utf8               | utf8_general_ci      | UTF-8 Unicode         |      3 |
+| utf8mb4            | utf8mb4_general_ci   | UTF-8 Unicode         |      4 |
++--------------------|----------------------|-----------------------|--------+
+5 rows in set (0.00 sec)
+```
+
+## COLLATIONS Table
+
+ `COLLATIONS` 表提供 `CHARACTER_SETS` 表中字符集对应的排序规则列表。因为 TiDB 当前仅支持二进制排序规则，包含该表仅为与 MySQL 保持一致。
```
Comment (lilin90) [inline sql/information-schema.md]:
```suggestion
 `COLLATIONS` 表提供了 `CHARACTER_SETS` 表中字符集对应的排序规则列表。TiDB 当前仅支持二进制排序规则，包含该表仅为兼容 MySQL。
```

File: sql/information-schema.md [hunk 3]
```diff
@@ -0,0 +1,441 @@
...
+| utf8mb4_croatian_ci    | utf8mb4            |  245 |            | Yes         |       1 |
+| utf8mb4_unicode_520_ci | utf8mb4            |  246 |            | Yes         |       1 |
+| utf8mb4_vietnamese_ci  | utf8mb4            |  247 |            | Yes         |       1 |
++------------------------+--------------------+------+------------+-------------+---------+
+26 rows in set (0.00 sec)
+```
+
+## COLLATION\_CHARACTER\_SET\_APPLICABILITY Table
+
+`COLLATION_CHARACTER_SET_APPLICABILITY` 表将排序规则映射给适用的字符集名称。和 `COLLATIONS` 表一样，包含此表也是为了和 MySQL 保持一致。
```
Comment (lilin90) [inline sql/information-schema.md]:
```suggestion
`COLLATION_CHARACTER_SET_APPLICABILITY` 表将排序规则映射至适用的字符集名称。和 `COLLATIONS` 表一样，包含此表也是为了兼容 MySQL。
```

File: sql/information-schema.md [hunk 4]
```diff
@@ -0,0 +1,441 @@
...
+| utf8mb4_croatian_ci    | utf8mb4            |
+| utf8mb4_unicode_520_ci | utf8mb4            |
+| utf8mb4_vietnamese_ci  | utf8mb4            |
++------------------------+--------------------+
+26 rows in set (0.00 sec)
+```
+
+## COLUMNS Table
+
+COLUMNS 表提供了关于所有表的列的信息。
```
Comment (lilin90) [inline sql/information-schema.md]:
```suggestion
COLUMNS 表提供了表的所有列的信息。
```

File: sql/information-schema.md [hunk 5]
```diff
@@ -0,0 +1,441 @@
...
+             COLUMN_TYPE: int(11)
+              COLUMN_KEY: 
+                   EXTRA: 
+              PRIVILEGES: select,insert,update,references
+          COLUMN_COMMENT: 
+   GENERATION_EXPRESSION: 
+1 row in set (0.01 sec)
+```
+
+The corresponding `SHOW` statement is as follows:
```
Comment (lilin90) [inline sql/information-schema.md]:
```suggestion
对应的 `SHOW` 语句如下：
```

File: sql/information-schema.md [hunk 6]
```diff
@@ -0,0 +1,441 @@
...
+| Field | Type    | Null | Key  | Default | Extra |
++-------+---------+------+------+---------+-------+
+| a     | int(11) | YES  |      | NULL    |       |
++-------+---------+------+------+---------+-------+
+1 row in set (0.00 sec)
+```
+
+## ENGINES Table
+
+`ENGINES` 表提供了关于存储引擎的信息。从和 MySQL 兼容性上考虑，TiDB 一直将 InnoDB 描述为唯一支持的引擎。
```
Comment (lilin90) [inline sql/information-schema.md]:
```suggestion
`ENGINES` 表提供了关于存储引擎的信息。从和 MySQL 兼容性上考虑，TiDB 会一直将 InnoDB 描述为唯一支持的引擎。
```

Comment (lilin90) [reply sql/information-schema.md]: Please ignore the previous comment and see the one below (the `Delete` button does not work currently):

Comment (lilin90) [inline sql/information-schema.md]:
```suggestion
`ENGINES` 表提供了关于存储引擎的信息。从和 MySQL 兼容性上考虑，TiDB 会一直称 InnoDB 为唯一支持的引擎。
```

File: sql/information-schema.md [hunk 7]
```diff
@@ -0,0 +1,441 @@
...
+     COMMENT: Supports transactions, row-level locking, and foreign keys
+TRANSACTIONS: YES
+          XA: YES
+  SAVEPOINTS: YES
+1 row in set (0.00 sec)
+```
+
+## KEY\_COLUMN\_USAGE Table
+
+`KEY_COLUMN_USAGE` 这张表描述了列的键约束，比如主键约束。
```
Comment (lilin90) [inline sql/information-schema.md]:
```suggestion
`KEY_COLUMN_USAGE` 表描述了列的键约束，比如主键约束。
```

File: sql/information-schema.md [hunk 8]
```diff
@@ -0,0 +1,441 @@
...
+POSITION_IN_UNIQUE_CONSTRAINT: NULL
+      REFERENCED_TABLE_SCHEMA: NULL
+        REFERENCED_TABLE_NAME: NULL
+       REFERENCED_COLUMN_NAME: NULL
+2 rows in set (0.00 sec)
+```
+
+## SCHEMATA Table
+
+SCHEMATA 表提供了关于数据库的信息。表中的内容和 `SHOW DATABASES` 语句执行结果基本等价。
```
Comment (lilin90) [inline sql/information-schema.md]:
```suggestion
SCHEMATA 表提供了关于数据库的信息。表中的数据与 `SHOW DATABASES` 语句的执行结果等价。
```

File: sql/information-schema.md [hunk 9]
```diff
@@ -0,0 +1,441 @@
...
+
+```sql
+mysql> select * from SCHEMATA;
++--------------|--------------------|----------------------------|------------------------|----------+
```
Comment (lilin90) [inline sql/information-schema.md]: The `DEFAULT_CHARACTER_SET_NAME` and `DEFAULT_COLLATION_NAME` are not the same as that in [the English version](https://github.com/pingcap/docs/pull/994/files). Please help check which one is more accurate.

File: sql/information-schema.md [hunk 10]
```diff
@@ -0,0 +1,441 @@
...
+| session_track_gtids              | OFF                  |
+| have_ssl                         | DISABLED             |
+| max_binlog_cache_size            | 18446744073709547520 |
++----------------------------------+----------------------+
+10 rows in set (0.00 sec)
+```
+
+## STATISTICS Table
+
+ `STATISTICS` 表提供关于表索引的信息。
```
Comment (lilin90) [inline sql/information-schema.md]:
```suggestion
 `STATISTICS` 表提供了关于表索引的信息。
```

File: sql/information-schema.md [hunk 11]
```diff
@@ -0,0 +1,441 @@
...
+   CONSTRAINT_NAME: delete_range_done_index
+      TABLE_SCHEMA: mysql
+        TABLE_NAME: gc_delete_range_done
+   CONSTRAINT_TYPE: UNIQUE
+6 rows in set (0.00 sec)
+```
+
+其中：
+
+* `CONSTRAINT_TYPE` 的取值可以是 `UNIQUE`, `PRIMARY KEY`, 或者 `FOREIGN KEY`。
```
Comment (lilin90) [inline sql/information-schema.md]:
```suggestion
* `CONSTRAINT_TYPE` 的取值可以是 `UNIQUE`，`PRIMARY KEY`，或者 `FOREIGN KEY`。
```

File: sql/information-schema.md [hunk 12]
```diff
@@ -0,0 +1,441 @@
...
+* `UNIQUE` 和 `PRIMARY KEY` 信息跟 `SHOW INDEX` 看到的是一样的。
```
Comment (lilin90) [inline sql/information-schema.md]:
```suggestion
* `UNIQUE` 和 `PRIMARY KEY` 信息与 `SHOW INDEX` 语句的执行结果类似。
```

File: sql/information-schema.md [hunk 13]
```diff
@@ -0,0 +1,441 @@
...
+
+## USER\_PRIVILEGES Table
+
+USER\_PRIVILEGES 表提供了关于全局权限的信息。这张表的内容是根据 `mysql.user` 表生成的。
```
Comment (lilin90) [inline sql/information-schema.md]:
```suggestion
USER\_PRIVILEGES 表提供了关于全局权限的信息。该表的数据根据 `mysql.user` 系统表生成。
```

File: sql/information-schema.md [hunk 14]
```diff
@@ -0,0 +1,441 @@
...
+| TABLE_CATALOG  | varchar(512) | YES  |      | NULL    |       |
+| PRIVILEGE_TYPE | varchar(64)  | YES  |      | NULL    |       |
+| IS_GRANTABLE   | varchar(3)   | YES  |      | NULL    |       |
++----------------|--------------|------|------|---------|-------+
+4 rows in set (0.00 sec)
+```
+
+## VIEWS Table
+
+`VIEWS` 表提供关于 SQL 视图的信息。
```
Comment (lilin90) [inline sql/information-schema.md]:
```suggestion
`VIEWS` 表提供了关于 SQL 视图的信息。
```

File: sql/system-database.md [hunk 15]
```diff
@@ -7,6 +7,8 @@ category: user guide
 
 TiDB 的系统数据库跟 MySQL 类似，里面包含一些服务器运行时需要的信息。
```
[the remark this answers is not in the document]
Comment (lilin90) [reply sql/system-database.md]: Please help fix this.

File: sql/information-schema.md (hunk 9 above)
Comment (lilin90) [reply sql/information-schema.md]: @tiancaiamao PTAL

File: sql/information-schema.md [hunk 16]
```diff
@@ -0,0 +1,225 @@
...
+
+INFORMATION\_SCHEMA 库里面的表主要是为了兼容 MySQL 而存在，有些第三方软件会查询里面的信息。在目前 TiDB 的实现中，里面大部分只是一些空表。
+
+## 支持 Information Schema 表
+
+### CHARACTER\_SETS Table
+
+提供字符集相关的信息，其实数据是假的。TiDB 默认支持并且只支持 utf8mb4 。
+
+```sql
```
[the remark this answers is not in the document]
Comment (morgo) [reply sql/information-schema.md]: This one is out of date.  The up to date version is:
```
mysql> SELECT * FROM character_sets;
+--------------------+----------------------+---------------+--------+
| CHARACTER_SET_NAME | DEFAULT_COLLATE_NAME | DESCRIPTION   | MAXLEN |
+--------------------+----------------------+---------------+--------+
| utf8               | utf8_bin             | UTF-8 Unicode |      3 |
| utf8mb4            | utf8mb4_bin          | UTF-8 Unicode |      4 |
| ascii              | ascii_bin            | US ASCII      |      1 |
| latin1             | latin1_bin           | Latin1        |      1 |
| binary             | binary               | binary        |      1 |
+--------------------+----------------------+---------------+--------+
5 rows in set (0.00 sec)
```

File: sql/information-schema.md (hunk 9 above)
Comment (morgo) [reply sql/information-schema.md]: This is the accurate version. ``` 
mysql> SELECT * FROM schemata;
+--------------+--------------------+----------------------------+------------------------+----------+
| CATALOG_NAME | SCHEMA_NAME        | DEFAULT_CHARACTER_SET_NAME | DEFAULT_COLLATION_NAME | SQL_PATH |
+--------------+--------------------+----------------------------+------------------------+----------+
| def          | INFORMATION_SCHEMA | utf8mb4                    | utf8mb4_bin            | NULL     |
| def          | mynewdb            | utf8mb4                    | utf8mb4_bin            | NULL     |
| def          | mysql              | utf8mb4                    | utf8mb4_bin            | NULL     |
| def          | PERFORMANCE_SCHEMA | utf8mb4                    | utf8mb4_bin            | NULL     |
| def          | test               | utf8mb4                    | utf8mb4_bin            | NULL     |
+--------------+--------------------+----------------------------+------------------------+----------+
5 rows in set (0.00 sec)```

File: sql/information-schema.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (lilin90) [reply sql/information-schema.md]: @morgo Got it. Thanks!

**Pull request merged**